File: plugins/eisen-architecture/skills/ports-and-adapters/templates/python/fake.py
"""FakeRegistrySource — an in-memory RegistrySource that ships with the package.

Lives in adapters/outbound/fakes/registry_source.py, next to the real adapters, so the
composition root, a ``--registry fake`` CLI flag and every test can use it. It is
deterministic, records each call, and can be told to fail with a domain error — which is how
tests exercise the port's documented failure behaviour without a network.
"""

from app.domain.contract import LoadRegistryRequest, LoadRegistryResult
from app.domain.errors import AppError


class FakeRegistrySource:
    """Satisfies ``RegistrySource`` from a canned result or a canned failure."""

    def __init__(self, result: LoadRegistryResult | None = None, *, failure: AppError | None = None) -> None:
        self.result = result or LoadRegistryResult()
        self.failure = failure
        self.calls: list[LoadRegistryRequest] = []

    def load(self, request: LoadRegistryRequest) -> LoadRegistryResult:
        """Record the request, then return the canned result or raise the canned failure."""
        self.calls.append(request)
        if self.failure is not None:
            raise self.failure
        return self.result

File: plugins/eisen-architecture/skills/ports-and-adapters/templates/typescript/fake.ts
/**
 * FakeRegistrySource — an in-memory RegistrySource that ships with the package.
 * Lives in adapters/outbound/fakes/registry-source.ts, next to the real adapters, so the
 * composition root, a `--registry fake` CLI flag and every test can use it. It is
 * deterministic, records each call, and can be told to fail with a domain error — which is
 * how tests exercise the port's documented failure behaviour without a network.
 */
import type { RegistrySource } from '../../../application/ports/registry-source';
import { CONTRACT_VERSION, type LoadRegistryRequest, type LoadRegistryResult } from '../../../domain/contract';
import type { AppError } from '../../../domain/errors';

export class FakeRegistrySource implements RegistrySource {
  readonly calls: LoadRegistryRequest[] = [];

  constructor(
    private readonly result: LoadRegistryResult = { contractVersion: CONTRACT_VERSION, apps: [] },
    private readonly failure: AppError | null = null,
  ) {}

  /** Record the request, then resolve the canned result or reject with the canned failure. */
  async load(request: LoadRegistryRequest): Promise<LoadRegistryResult> {
    this.calls.push(request);
    if (this.failure !== null) throw this.failure;
    return this.result;
  }
}
